From: Romain Janvier <romain.janvier@hotmail.fr>
Date: Wed, 23 Jan 2013 10:20:38 +0100
Subject: Remove-pmvs-dir-is-default

Application now remove pmvs dir if it exists in
the target directory
---
 osmcmvs/__init__.py |    8 ++++++--
 osmpmvs/__init__.py |    8 ++++++--
 2 files changed, 12 insertions(+), 4 deletions(-)

--- a/osmcmvs/__init__.py
+++ b/osmcmvs/__init__.py
@@ -1,5 +1,5 @@
 import logging
-import sys, os, getopt, subprocess, shutil
+import sys, os, getopt, subprocess, shutil, tempfile
 
 # service function: get path of an executable (.exe suffix is added if we are on Windows)
 def getExecPath(dir, fileName):
@@ -69,8 +69,17 @@ class OsmCmvs():
         # just run Bundle2PMVS here
         logging.info("\nPerforming Bundler2PMVS conversion...")
         os.chdir(self.workDir)
-        os.mkdir("pmvs")
-
+        try:
+            os.mkdir("pmvs")
+        # New behavior: move old pmvs dir
+        except OSError, e:
+            if e.errno != 17:
+                raise
+            print "pmvs dir already exists in " + os.getcwd() + "..." 
+            cpydir = tempfile.mkdtemp(prefix="pmvs-", dir=os.getcwd())
+            print "...moving it to " + cpydir
+            os.renames("pmvs", cpydir)
+            os.mkdir("pmvs")
         # Create directory structure
         os.mkdir("pmvs/txt")
         os.mkdir("pmvs/visualize")
--- a/osmpmvs/__init__.py
+++ b/osmpmvs/__init__.py
@@ -63,8 +63,17 @@ class OsmPmvs():
         # just run Bundle2PMVS here
         logging.info("\nPerforming Bundler2PMVS conversion...")
         os.chdir(self.workDir)
-        os.mkdir("pmvs")
-
+        try:
+            os.mkdir("pmvs")
+        # New behavior: move old pmvs dir
+        except OSError, e:
+            if e.errno != 17:
+                raise
+            print "pmvs dir already exists in " + os.getcwd() + "..." 
+            cpydir = tempfile.mkdtemp(prefix="pmvs-", dir=os.getcwd())
+            print "...moving it to " + cpydir
+            os.renames("pmvs", cpydir)
+            os.mkdir("pmvs")
         # Create directory structure
         os.mkdir("pmvs/txt")
         os.mkdir("pmvs/visualize")
